ddtrace/propagation/http.py

```diff
@@ -483,6 +487,102 @@ def _extract(headers):
+      value           = version "-" version-format
+      version         = 2HEXDIGLC
+      version-format  = trace-id "-" parent-id "-" trace-flags
+      trace-id        = 32HEXDIGLC
+      parent-id       = 16HEXDIGLC
+      trace-flags     = 2HEXDIGLC
+
+    Example value of HTTP ``traceparent`` header::
+
+        value = 00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01
+        base16(version) = 00
+        base16(trace-id) = 4bf92f3577b34da6a3ce929d0e0e4736
+        base16(parent-id) = 00f067aa0ba902b7
+        base16(trace-flags) = 01  // sampled
+
+    Implementation details:
+
+      - Datadog Trace and Span IDs are 64-bit unsigned integers.
+      - The W3C Trace Context Trace ID is a 32-byte hexadecimal string. This is
+        transformed for propagation between Datadog by taking the lower
+        16-bytes.
+    """
+
+    @staticmethod
+    def _inject(span_context, headers):
+        # type: (Context, Dict[str, str]) -> None
+        # use hex to convert back, the trace id will be pre-pended with a bunch of 0s to fill in for previous values
+        trace_id = span_context.trace_id
+        span_id = span_context.span_id
+        if trace_id is None or span_id is None:
+            log.debug("tried to inject invalid context %r", span_context)
+            return
+
+        sampling_priority = span_context.sampling_priority
+        sampling_priority_hex = "01" if sampling_priority and sampling_priority >= AUTO_KEEP else "00"
+
+        # There is currently only a single version so we always start with 00
+        traceparent = "{}-{}-{}-{}".format(
+            "00",
+            "0000000000000000" + "%x" % trace_id,  # we use %x to avoid hex() prepend and append
+            "%x" % span_id,
+            sampling_priority_hex,
+        )
+        headers[_HTTP_HEADER_W3C_TRACEPARENT] = traceparent
+
+    @staticmethod
+    def _extract(headers):
+        # type: (Dict[str, str]) -> Optional[Context]
+        tp = _extract_header_value(_POSSIBLE_HTTP_HEADER_W3C_TRACEPARENT, headers)
+        if tp is None:
+            return None
+
+        try:
+            version, trace_id_hex, span_id_hex, trace_flags = tp.split("-")
+        except AttributeError:
+            log.warning(
+                "Trace context will not be propagated because Traceparent:%s is not a string, type:%s", (tp, type(tp))
+            )
+        try:
+            # currently 00 is the only version format, but if future versions come up we may need to add changes
+            if version == "00" and len(trace_id_hex) == 32 and len(span_id_hex) == 16 and len(trace_flags) == 2:
+                trace_id = int(trace_id_hex[-16:], 16 & 0xFFFFFFFFFFFF)
+                span_id = int(span_id_hex, 16 & 0xFFFFFFFFFFFF)
+                # there's only one trace flag, which denotes sampling priority was set to keep
+                sampling_priority = AUTO_KEEP if trace_flags == "01" else AUTO_REJECT
```

Comment:
Let's bitmask off the specific sampling portion of the trace_flags. In the case any other trace flags are set else the sampling extraction logic here will be broken.